src/apps/ENGINE/file_service.cpp

```diff
@@ -40,53 +41,54 @@ FILE_SERVICE::~FILE_SERVICE()
     Close();
 }
 
-HANDLE FILE_SERVICE::_CreateFile(const char *lpFileName, uint32_t dwDesiriedAccess, uint32_t dwShareMode,
-                                 uint32_t dwCreationDisposition)
+std::fstream FILE_SERVICE::_CreateFile(const char *filename, std::ios::openmode mode)
 {
-    HANDLE fh;
-    std::wstring filePathW = utf8::ConvertUtf8ToWide(lpFileName);
-    fh = CreateFile(filePathW.c_str(), dwDesiriedAccess, dwShareMode, nullptr, dwCreationDisposition,
-                    FILE_ATTRIBUTE_NORMAL, nullptr);
-    return fh;
+    std::fstream fileS(filename, mode);
+    return fileS;
 }
 
-void FILE_SERVICE::_CloseHandle(HANDLE hFile)
+void FILE_SERVICE::_CloseFile(std::fstream &fileS)
 {
-    CloseHandle(hFile);
+    fileS.close();
 }
 
-uint32_t FILE_SERVICE::_SetFilePointer(HANDLE hFile, long DistanceToMove, long *lpDistanceToMoveHigh,
-                                       uint32_t dwMoveMethod)
+void FILE_SERVICE::_SetFilePointer(std::fstream &fileS, std::streamoff off, std::ios::seekdir dir)
 {
-    return SetFilePointer(hFile, DistanceToMove, lpDistanceToMoveHigh, dwMoveMethod);
+    fileS.seekp(off, dir);
 }
 
-BOOL FILE_SERVICE::_DeleteFile(const char *lpFileName)
+int FILE_SERVICE::_DeleteFile(const char *filename)
 {
-    std::wstring filePathW = utf8::ConvertUtf8ToWide(lpFileName);
-    return DeleteFile(filePathW.c_str());
+    std::filesystem::path path = std::filesystem::u8path(filename);
+    return std::remove(filename);
```

Comment:
you removing 'filename' instead of 'path'
I suppose you meant `std::filesystem::remove(path);`

Reply:
Good catch. Thanks!